ni_measurementlink_service/_internal/parameter/serializer.py

```diff
@@ -67,6 +75,16 @@ def serialize_parameters(
             parameter_metadata.type,
             parameter_metadata.repeated,
         )
+        # Convert enum parameters to their underlying value.
+        if "ni/type_specialization" in parameter_metadata.annotations:
+            if (
+                parameter_metadata.annotations["ni/type_specialization"]
+                == TypeSpecialization.Enum.value
+            ):
+                if parameter_metadata.repeated:
+                    parameter = list(map(lambda x: x.value, parameter))
```

Comment:
List comprehensions are preferred.
```suggestion
                    parameter = [x.value for x in parameter]
```